src/bash-completion.sh

```diff
@@ -23,8 +22,46 @@ _strip()
 	fi
 }
 
+_get_idx_files()
+{
+	local d=/var/cache/zypp/solv
+	local idx=$d/*/solv.idx
+	# this part covers the case
+	# where no idx files have been created by libzypp yet
+	# TODO: drop in 2019
+        set +o noglob
+	if echo $idx | grep -q -F '*' ; then
```

Comment:
You can't silently remove the solv files.
Existing solv files may belong to volatile media (CD/DVD/USB), so you'd enforce zypper asking to insert the media again. I can try to create missing idx files in zypp even if the solv file is up-to-date.